### ISSUE TITLE
throws inside check() do not need to specify the message key while throws inside coerce do

### ISSUE THREAD

**Issue by Linard11** [yargs/yargs#1172]
To make yargs fail I can simply throw inside a check() callback function with:
`throw 'check failed!'` and yargs will print this string as the YError, but when throwing inside a coerce callback function the same throw will result in the generic `yargs error` and I have to a throw as `throw { message: 'coerce failed!}` in order to see my error message on the console.

**Comment by xmedeko** [comment]
This is working with the current `17.7.2` version.

**Comment by shadowspawn** [comment]
There is indeed specific code in `check` to cope with throwing a string: https://github.com/yargs/yargs/blob/e517318cea0087b813f5de414b3cdec7b70efe33/lib/yargs-factory.ts#L324

> and I have to a throw as `throw { message: 'coerce failed!}`

I recommend you throw an `Error`. This has a message property which lots of error handling code expects, and will work with both `check` and `coerce`. 

So your code would look like:

```
throw new Error('coerce failed!')
```

See for example recommendation to throw an error here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw#description

**Comment by xmedeko** [comment]
Yeah, that's it: in `check`, it's possible to throw string, while in coerce it's not. `yargs` should be consistent in this matter, because `coerce` may do checking, too.

**Comment by shadowspawn** [comment]
I recommend throwing an `Error`. I don't want to extend the behaviour that `check` allows throwing a string.
